docs/reference/disallow-pulling.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: 'Disallow pulling Gadgets'
+sidebar_position: 600
+description: How to disallow pulling gadgets
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+It is possible to disallow pulling gadgets by using the `--disallow-pulling` flag.
+By default, pulling is allowed.
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+You can specify this option only at deploy time:
+
+```bash
+$ kubectl gadget deploy --disallow-gadgets-pulling
+...
+Inspektor Gadget successfully deployed
+$ kubectl gadget run trace_exec
+Error: fetching gadget information: getting gadget info: rpc error: code = Unknown desc = getting gadget info: initializing and preparing operators: instantiating operator "oci": ensuring image: pulling image (if missing) "trace_exec": pulling image "trace_exec": pulling is disabled
+```
```

Comment:
We for sure can document it, but I think this option is useless until we provide a way to provision images or implement something like #2417.